### PR TITLE
feat: Allow unsafeSsl for artemis connections

### PR DESCRIPTION
**Pull request by NoUseFreak**
Add `unsafeSsl` option (like the prometheus scaler) to allow self-signed certificates.

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) -> https://github.com/kedacore/keda-docs/pull/1042
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #

Relates to #

### CODE REVIEW

File: pkg/scalers/artemis_scaler.go [hunk 1]
```diff
@@ -208,6 +193,44 @@ func getAPIParameters(meta artemisMetadata) (artemisMetadata, error) {
 	return meta, nil
 }
 
+func parseArtemisAuth(config *ScalerConfig) (*authentication.AuthMeta, error) {
```
Comment (JorTurFer) [inline pkg/scalers/artemis_scaler.go:196]: As you have changed to `AuthMeta`, there is a helper to parse it `GetAuthConfigs` with all the options (event the scaler doesn't support all, we can use the helper and ignore invalid options)

File: pkg/scalers/artemis_scaler.go [hunk 2]
```diff
@@ -173,6 +174,16 @@ func parseArtemisMetadata(config *ScalerConfig) (*artemisMetadata, error) {
 
 	meta.scalerIndex = config.ScalerIndex
 
+	meta.unsafeSsl = false
+	if val, ok := config.TriggerMetadata[unsafeSsl]; ok && val != "" {
+		unsafeSslValue, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", unsafeSsl, err)
```
[the remark this answers is not in the document]
Comment (JorTurFer) [reply pkg/scalers/artemis_scaler.go:155]:
```suggestion
			return nil, fmt.Errorf("error parsing %s: %w", unsafeSsl, err)
```

**Review by JorTurFer — COMMENTED** [review]
Looking good, only 2 small nits inline

**Review by JorTurFer — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -58,6 +58,7 @@ Here is an overview of all new **experimental** features:
 
 ### Improvements
 
+- **Artemis**: Add unsafeSsl option ([#4081](https://github.com/kedacore/keda/pull/4081))
```
Comment (zroubalik) [inline CHANGELOG.md:61]: please move this below "General" entries

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -61,6 +61,7 @@ Here is an overview of all new **experimental** features:
 - **General**: Use (self-signed) certificates for all the communications (internals and externals) ([#3931](https://github.com/kedacore/keda/issues/3931))
 - **Redis Scalers**: Add support to Redis 7 ([#4052](https://github.com/kedacore/keda/issues/4052))
 - **Selenium Grid Scaler**: Add 'platformName' to selenium-grid scaler metadata structure ([#4038](https://github.com/kedacore/keda/issues/4038))
+- **Artemis**: Add unsafeSsl option ([#4081](https://github.com/kedacore/keda/pull/4081))
```
Comment (zroubalik) [inline CHANGELOG.md:64]:
```suggestion
- **Artemis Scaler**: Add unsafeSsl option ([#4081](https://github.com/kedacore/keda/pull/4081))
```

**Comment by JorTurFer** [comment]
Does `unsafeSsl` make sense once the [customCA](https://github.com/kedacore/keda/pull/4191) is merged?

**Comment by NoUseFreak** [comment]
Closing as customCA can solve the issue

**Pull request closed without merging**